python/lsst/verify/extract_metricvalues.py

```diff
@@ -149,20 +151,27 @@ def _match_data_id(data_id, data_id_restriction):
     if data_id_restriction is None:
         return True
     for key, value in data_id_restriction.items():
-        if not (data_id[key] == value):
+        if key not in data_id or not (data_id[key] == value):
```

Comment:
I realize this is prior code, but I think `data_id[key] != value` would be a lot clearer than `not (data_id[key] == value)`.